### PR TITLE
Ref #7850. Fix compatibility checking between two aliases [7851]

### PR DESCRIPTION
**Pull request by jamoralp**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/111/)

File: include/xtypes/AliasType.hpp [hunk 1]
```diff
@@ -105,7 +105,18 @@ class AliasType : public DynamicType
 
     virtual TypeConsistency is_compatible(const DynamicType& other) const override
     {
-        return aliased_->is_compatible(other);
+        TypeConsistency consistency = TypeConsistency::EQUALS;
+        if (other.kind() == TypeKind::ALIAS_TYPE)
+        {
+            const AliasType& other_alias = static_cast<const AliasType&>(other);
+            consistency |= aliased_->is_compatible(other_alias.rget());
```
Comment (LuisGP) [inline include/xtypes/AliasType.hpp]:
```suggestion
            consistency |= rget().is_compatible(other_alias.rget());
```

File: include/xtypes/AliasType.hpp [hunk 2]
```diff
@@ -105,7 +105,18 @@ class AliasType : public DynamicType
...
+        }
+        else
+        {
+            consistency = aliased_->is_compatible(other);
```
Comment (LuisGP) [inline include/xtypes/AliasType.hpp]:
```suggestion
            consistency = rget().is_compatible(other);
```

**Review by LuisGP — COMMENTED** [review]
What do you think about the suggestions? I think they may improve the performance a little.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/112/)

**Review by LuisGP — APPROVED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/113/)

**Pull request merged**